### PR TITLE
Enable remote authentication with HTTP Signatures

### PR DESCRIPTION
**Pull request by annando**
This is a follow up from PR https://github.com/friendica/friendica/pull/6906

With this PR we might have solved one of the problems when displaying non public photos.

### CODE REVIEW

File: src/Model/Profile.php [hunk 1]
```diff
@@ -1133,6 +1119,43 @@ public static function openWebAuthInit($token)
 
 			$_SESSION['remote'][] = ['cid' => $contact['id'], 'uid' => $contact['uid'], 'url' => $visitor['url']];
 		}
+
+		$a->contact = $visitor;
```
Comment (nupplaphil) [inline src/Model/Profile.php]: As we currently call this method out of app, I'd prefer returning the visitor and set it as `$this->contact` inside `App`.

**Comment by nupplaphil** [comment]
I'm not sure about adding new functionality to the RC branch, since there are possible regressions included :-). Wouldn't it better to add it to the develop branch, would it?

**Comment by annando** [comment]
@nupplaphil This is part of a fix for the problem that another contact can't see our private pictures. Although this also is a preparation for upcoming stuff, I tried to keep this as easy as possible.

File: src/Model/Profile.php (hunk 1 above)
Comment (annando) [reply src/Model/Profile.php]: This is just a copy from the already existing code from the function "openWebAuthInit" in this file. I wanted to only apply the needed changes and not to restructure everything.

Comment (nupplaphil) [reply src/Model/Profile.php]: I know this kind of problem ;-)
OK, at least I'd like to add a `@todo` for later. This is something I'd like to generally avoid in the code
`App` -> `AnotherClass` -> `App` dependency.

**Comment by nupplaphil** [comment]
OK, I see

File: src/Model/Profile.php (hunk 1 above)
Comment (annando) [reply src/Model/Profile.php]: BTW: This is some other "I don't really understand the problem" thing ;-) Have I said that you have to explain me more of this in some meeting, possibly at the Hackathon in Berlin? :grinning:

Comment (nupplaphil) [reply src/Model/Profile.php]: I'll do my best to join at the Hackathon :-)

**Pull request merged**